spec/account_spec.rb

```diff
@@ -0,0 +1,98 @@
+require "spec_helper"
+require "bankaccount"
+
+describe BankAccount::Account do
+	
+	it "responds to class method called new" do
+		expect(BankAccount::Account).to respond_to :new
+	end
+
+	describe "it creates an instance of a new Account with an id and a positive initial_balance" do
+		let(:account) { BankAccount::Account.new(1, 500) }
+
+		it "has the id 1" do
+			expect(account.id).to eq 1
+		end
+
+		it "has an initial_balance of 500" do
+			expect(account.initial_balance).to eq 500
+		end
+
+		it "raises an error if a negative initial_balance is entered" do
+			expect{BankAccount::Account.new(4, -400)}.to raise_exception(ArgumentError)
+		end
+	end
+
+	describe "an Account instance with an initial balance of 5000" do
+		let(:account) { BankAccount::Account.new(1, 5000) }
+
+		describe "you can withdraw money from your account" do
+			it "responds to the method 'withdraw'" do
+				expect(account).to respond_to :withdraw
+			end
+
+			context "reduces the account balance when the withdraw method is called" do
+				it "returns 4000 when you withdraw 1000" do
+					expect(account.withdraw(1000)).to eq 4000
+				end
+
+				it "returns 3500 when you withdraw 1000 and then withdraw 500" do
```

Comment:
It is not immediately obvious why you would want this test as the concept is covered by the previous test. Is it testing that you can chain the method? Or is it testing that the return state of withdraw changes the balance in a way that persists? This would be good to clarify in the it string. If it is the chaining, that falls into the category of things we don't need to test because we can trust the ruby core team to have tested core ruby functionality. The persistence bit might be worth testing if the implementation changed and no longer persisted you'd want your tests to alert you about it.